hail/python/hail/table.py

```diff
@@ -3493,7 +3493,13 @@ def multi_way_zip_join(tables, data_field_name, global_field_name) -> 'Table':
             [t._tir for t in tables], data_field_name, global_field_name))
 
     def _group_within_partitions(self, name, n):
-        return Table(ir.TableGroupWithinPartitions(self._tir, name, n))
+        def grouping_func(part):
+            groups = part.grouped(n)
+            key_names = list(self.key)
+            return groups.map(lambda group:
+                              hl.struct(**{key_name: group[0][key_name] for key_name in key_names}).annotate(**{name: group}))
```

Comment:
This is probably better as `group[0].select(*key_names, **{name: group})` (I think that works)

Reply:
Nice, that is better.